docs/plugins/README.md

```diff
@@ -0,0 +1,47 @@
+# Plugin Docs
+
+* [Installation](#installation)
+* [Compiling Plugins](#compiling-plugins)
+* [Registering Plugin](#registering-plugin)
+* [Creating ConfigMap](#creating-configmap)
+
+### Installation
+
+To enable plugin support in the dashboard you must register a custom [CRD](../../aio/test-resources/plugin-crd.yml) in your cluster.
+```
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/master/aio/test-resources/plugin-crd.yml
+```
+
+### Compiling Plugins
+
+In order to take advantage of AOT compilation, sharing the code across plugins and not ship the core Angular packages with the plugin bundle, there is a custom build process to compile the plugins.
+You can clone this repository and checkout to [`plugin/base`](https://github.com/kubernetes/dashboard/tree/plugin/base) branch. The build process is specified under [`builders`](https://github.com/kubernetes/dashboard/tree/plugin/base/builders) directory.
+
+On this branch, you can compile the plugin with following command
+```
+ng build plugin && ng build --project custom-plugin --prod --modulePath=k8s-plugin#PluginModule --pluginName=k8s-plugin --outputPath=./dist/bundle
+ng build --project custom-plugin --prod --modulePath=./plugin1/plugin1.module#Plugin1Module --pluginName=plugin1 --sharedLibs=k8s-plugin --outputPath=./dist/bundle
+```
+
+The key thing here is that we specify the `custom-plugin` project in the `angular.json` to use our custom builder. Make sure to keep the config similar when developing your own plugins.
+
+### Registering Plugin
+
+Once the custom CRD is registered we can now create [instances](../../aio/test-resources/plugin-test.yml) of the CRD which will hold the spec for plugin.
+
+```
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/master/aio/test-resources/plugin-test.yml
+```
+
+> Note: The backend reads the compiled plugin source from a ConfigMap and we need to create that also.
+
+### Creating ConfigMap
```

Comment:
kubernetes configmap has size limitation, (<=1MB), so the compiled plugin javascript file should not exceed 1MB. That is the limitation of this plugin solution, right?

Reply:
Yes, you're right. We can look into other ways to obtain the compiled plugin file instead of just configmap. We wanted to keep dashboard stateless and using configmaps was the easiest way to tackle this for now.